pkg/backend/display/progress.go

```diff
@@ -666,6 +666,8 @@ func (display *ProgressDisplay) processEndSteps() {
 	wroteDiagnosticHeader := false
 
 	for _, row := range display.eventUrnToResourceRow {
+		wroteResourceHeader := false
```

Comment:
this was the original intent of the code here.  We iterate over each resource so that all the messages are grouped by resource.  The intent was that if we then had any diagnostics to show for that resource, we would print out its name once, then all the diags.  but my logic was bad (or this got munged around over time), and we ended up printing the resource name again for each "stream" of messages for that specific resource.